src/DependencyInjection/DI/src/ServiceLookup/DynamicServiceProviderEngine.cs

```diff
@@ -19,12 +19,30 @@ public DynamicServiceProviderEngine(IEnumerable<ServiceDescriptor> serviceDescri
             var callCount = 0;
             return scope =>
             {
+                // Resolve the result before we increment the call count, this ensures that singletons
+                // won't cause any side effects during the compilation of the resolve function.
+                var result = RuntimeResolver.Resolve(callSite, scope);
+
                 if (Interlocked.Increment(ref callCount) == 2)
                 {
-                    Task.Run(() => base.RealizeService(callSite));
+                    // Don't capture the ExecutionContext when forking to build the compiled version of the
+                    // resolve function
+                    ThreadPool.UnsafeQueueUserWorkItem(state =>
+                    {
+                        try
+                        {
+                            base.RealizeService(callSite);
+                        }
+                        catch
+                        {
+                            // Swallow the exception, we should log this via the event source in a non-patched release
```

Comment:
Do we need an issue for this TODO? or will we take care of it when we port this change to 5.0+?

Reply:
I would take care of it in the port.